### PR TITLE
[DOC] Document additional analysis designs for non-parametric methods

### LINKED ISSUE #3159: `permuted_ols` and `non_parametric_inference` are not well documented
`permuted_ols` actually works for one-sample tests, which is not documented.
Also, the outputs are only -log_10( corrected_p_values), which is a bit limited too. It is also ill-documented.

### PR DESCRIPTION
**Pull request by tsalo**
Closes #3159. Instead of adding a new example that uses `permuted_ols` with a one-sample design, I added more documentation about designs and linked to an existing one-sample example using `non_parametric_inference` (which just wraps around `permuted_ols`).

Changes proposed in this pull request:

- Mention one-sample designs in documentation/examples related to `permuted_ols` and `non_parametric_inference`.

### CODE REVIEW

File: examples/05_glm_second_level/plot_second_level_one_sample_test.py [hunk 1]
```diff
@@ -62,8 +62,11 @@
 ###############################################################################
 # Estimate second level model
 # ---------------------------
-# We define the input maps and the design matrix for the second level model
-# and fit it.
+# We wish to perform a one-sample test.
+# In order to do so, we need to create a design matrix that determines how
+# the analysis will be performed.
+# For a one-sample test, all we need to include in the design matrix is a
+# single column of ones, corresponding to the model intercept.
```
Comment (bthirion) [inline examples/05_glm_second_level/plot_second_level_one_sample_test.py:69]: Sounds great. I would simply add:
the one-sample test assumes that the distribution is symmetric about 0 (which is is weaker than assuming that it is Gaussian and centered about 0).

Comment (tsalo) [reply examples/05_glm_second_level/plot_second_level_one_sample_test.py:69]: Are you referring to both `SecondLevelModel` and `permuted_ols`, or just one of them?

**Comment by tsalo** [comment]
@ymzayek sorry, I totally forgot to follow up on @bthirion's comment. I'll update once I understand the recommendation a bit better and can incorporate it without making things unclear.

File: examples/05_glm_second_level/plot_second_level_one_sample_test.py (hunk 1 above)
Comment (bthirion) [reply examples/05_glm_second_level/plot_second_level_one_sample_test.py:69]: Sorry, `permuted_ols`

Comment (tsalo) [reply examples/05_glm_second_level/plot_second_level_one_sample_test.py:69]: WDYT of this?

```python
# .. important::
#   One key difference between
#   :obj:`~nilearn.glm.second_level.SecondLevelModel` and
#   :func:`~nilearn.glm.second_level.non_parametric_inference`/
#   :func:`~nilearn.mass_univariate.permuted_ols`
#   is that the one-sample test in non_parametric_inference/permuted_ols
#   assumes that the distribution is symmetric about 0,
#   which is is weaker than the SecondLevelModel's assumption that
#   the null distribution is Gaussian and centered about 0.
```

Comment (bthirion) [reply examples/05_glm_second_level/plot_second_level_one_sample_test.py:69]: Yes, thx !

Comment (tsalo) [reply examples/05_glm_second_level/plot_second_level_one_sample_test.py:69]: I've added it. Thank you.

**Comment by tsalo** [comment]
@ymzayek I can add a whatsnew entry, but I'm not sure it's worth it. This is just a minor documentation update, after all. WDYT?

**Comment by ymzayek** [comment]
@tsalo sure in this case it's fine not to include one

**Review by bthirion — APPROVED** [review]
LGTM thx

File: doc/glm/second_level_model.rst [hunk 2]
```diff
@@ -51,7 +52,13 @@ important to address with statistical analysis of fMRI data. Nilearn provides pa
 to address this issue.
 
 Refer to the example :ref:`sphx_glr_auto_examples_05_glm_second_level_plot_thresholding.py` for a guide
-to applying FPR, FDR and FWER corrections. These corrections are applied using the :func:`nilearn.glm.threshold_stats_img` function.
+to applying FPR, FDR and FWER corrections.
```
Comment (NicolasGensollen) [inline doc/glm/second_level_model.rst:55]:
```suggestion
to applying :term:`FPR<FPR correction>`, :term:`FDR<FDR correction>`, and :term:`FWER<FWER correction>` corrections.
```

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @tsalo !
LGTM, I just have a small glossary suggestion.

File: doc/glm/second_level_model.rst (hunk 2 above)
Comment (tsalo) [reply doc/glm/second_level_model.rst:55]: Thanks @NicolasGensollen!

**Review by NicolasGensollen — APPROVED** [review]
LGTM, Thanks @tsalo ! :+1:

**Comment by tsalo** [comment]
Thanks all. Merging now.

**Pull request merged**